base/src/main/java/io/spine/code/proto/enrichment/EnrichmentForOption.java

```diff
@@ -126,7 +127,13 @@ protected MessageOptions optionsFrom(Descriptor object) {
      * Makes sure that if a passed type reference is direct reference to a type,
      * it is a fully-qualified reference, or becomes one as the result of this method.
      */
+    @SuppressWarnings("ChainOfInstanceofChecks") /* TypeRef interface does not provide
+                                                    the means to check the package. */
     private static TypeRef ensurePackage(PackageName packageName, TypeRef ref) {
+        if (ref instanceof InPackage) {
```

Comment:
Why do we need to cast? Why cannot we have a virtual `expand()` method that would return the same type, but with the fully-qualified reference to the same thing?

It could be a decorator per type, as we probably want to retain the original reference information.